src/commit-body.ts

```diff
@@ -1,21 +1,80 @@
+import Ajv, {JTDParser, JTDSchemaType} from 'ajv/dist/jtd'
+import {InvalidCommitBodyError} from './errors'
 import {Message} from './message'
 
+interface CommitBodyMetaData {
+  job_number?: number
+  job_commit?: string
+}
+
+interface CommitBodyData {
+  namespace: 'git-queue.commit-body'
+  version: number
+  metadata: CommitBodyMetaData
+  payload: string
+}
+
+const CommitBodySchema: JTDSchemaType<CommitBodyData> = {
+  properties: {
+    namespace: {enum: ['git-queue.commit-body']},
+    version: {type: 'int32'},
+    payload: {type: 'string'},
+    metadata: {
+      optionalProperties: {
+        job_number: {type: 'int32'},
+        job_commit: {type: 'string'}
+      },
+      additionalProperties: true
+    }
+  },
+  additionalProperties: true
+}
+
 export class CommitBody {
-  text: string
+  body: CommitBodyData
+  private static parse: JTDParser<CommitBodyData> = new Ajv().compileParser(
+    CommitBodySchema
+  )
 
   constructor(text: string) {
```

Comment:
Well, as the JTD (https://ajv.js.org/guide/typescript.html) is usually used, we use a JSON Data type (coupled to the underlying technology) in the structures definition, and a CommitBodySchema to represent this type with all needed annotations. So in the constructor, what I think is suitable, is the DataType, not the Schema.

Reply:
I had a discussion with @yeraydavidrodriguez, The object:

```
    {
      namespace: 'git-queue.commit-body',
      version: 1,
      metadata: {
        job_commit: message.getJobRef().getHash()
      },
      payload: message.getPayload()
    }
```

needs to be validated anyway, because the interface `CommitBodyData` is a superset of the schema, which means it would allow building objects that do not match the schema. And that validation is basically the main responsibility of CommitBody class. It guards that the data structure `CommitBodyData` matches the commit body schema.